src/jit/morph.cpp

```diff
@@ -3415,11 +3434,7 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
                     size = (unsigned)(roundUp(info.compCompHnd->getClassSize(argx->gtArgPlace.gtArgPlaceClsHnd),
                                               TARGET_POINTER_SIZE)) /
                            TARGET_POINTER_SIZE;
-                    if (isHfaArg)
-                    {
-                        hasMultiregStructArgs = true;
-                    }
-                    else if (size > 1 && size <= 4)
+                    if (isHfaArg || size > 1)
                     {
```

Comment:
Why does this need to change? Is this a bug in existing `LEGACY_BACKEND`? Or a new requirement for RyuJIT? (To not check `size <= 4`)